section2/reflection.md

```diff
@@ -1,15 +1,18 @@
-
+-Reflecting on the process of coding, go over what worked and what didn't.
+-Assess WHAT I am stuck on and then create an action plan to solve it.
 2. In which ways do you currently demonstrate a Growth Mindset? In which ways do you _not_?
-
+-I currently demonstrate growth mindset in that I believe I can learn anything I want to with hardwork and discipline.
+-I do _not_ in that I get discouraged easily or wonder where to even start in solving a problem.
 3. In this section, we removed some of the supports that you had in Section 1. We didn't give the directions for how to run a file in node and made some directions a bit more vague, intentionally. How did that feel? What was uncomfortable about it? How did it support your learning?
-
+At first it felt like I maybe missed something or didn't read carefully enough so I went back and reread. Once I decided I hadn't missed anything, I went to google to figure out how to run a file in node. It felt good that I was able to identify quickly what I needed to learn and solved it on my own. It supported my learning in giving me an opportunity to figure something out on my own before asking someone for help.
 4. What is a conditional statement? Give one example of a daily life conditional. Give one example of where a conditional is probably used in a web application you use.
-
+A conditional statement is a condition or parameter that is set and is either true or false. A daily life conditional could be wakeUpHungry. If true, then I eat. If false then nothing happens. A web application conditional could be when I enter my password to login to my email. If it matches then it is true and I am taken to the page with my emails. If it doesn't match then it is false and if...elsle is run and the page stays the same with the words "password is incorrect".
 5. How do you add multiple conditions to an `if` statement? In your own words, explain how to program reads them and determines what to do.
-
+To add multiple conditions to an 'if' statement you must have else if with the other option(s).The program reads the first conditional and if it is true it stops there and executes the message for that condition. If the first conditional is false then the program moves on to the second conditional. If that is false it moves on to the third, so on and so forth until whichever conditional IS true, that is what the program executes.
 6. What tools are available to you, as a developer, if you want to check that TWO conditions are met? Or, if you want to check that 1 of 2 conditions are met, before running a block of code?
-
-7. What questions do you still have about `if` statements and/or functions?
\ No newline at end of file
+Run the scenario in your terminal with node to see what it prints.
+7. What questions do you still have about `if` statements and/or functions?
+What instances would you use 'null' or 'undefined'?
```

Comment:
Both are pretty common. Sometimes `undefined` is the return value when your code doesn't work as intended.

Say I have an array:

```js
var arr = [1, 2, 3]
```

If I try to access the 4th element, what's the return value?

```js
arr[3]
// => undefined
```

So if I was writing code that used this array but I wasn't sure if there was a 4th element or not, I might write a condition that checks if `arr[4] === undefined` so I can do something different in that case.

Null...I wouldn't worry about too much for now but you'll get more exposure soon 😄